src/main/java/de/dentrassi/rpm/builder/RpmMojo.java

```diff
@@ -740,7 +740,13 @@ public void execute () throws MojoExecutionException, MojoFailureException
 
             if ( this.attach )
             {
-                this.projectHelper.attachArtifact ( this.project, "rpm", this.classifier, builder.getTargetFile ().toFile () );
+                this.logger.info ( "attaching %s", this.classifier);
+                if ( "rpm".equals( project.getPackaging() ) ){
+                    project.getArtifact().setFile( builder.getTargetFile().toFile() );
```

Comment:
Shouldn't this branch use `this.projectHelper.attachArtifact ( project, artifactFile, artifactClassifier );` instead of manually changing the project model?

Reply:
Apparently that confuses it because it tries to attach it as primary and thinks it has a primary.  This is the only way I've found that lets it treat it as the primary artifact

Reply:
I am sorry, but I don't understand that.

Reply:
so attaching the artifact is for secondaries.  when you NULL out the values to act like a primary artifact maven complains b/c it's not the right way to add the primary artifact

This was the only way I could find (I think I originally found it in the mvn-rpm-plugin source? or maybe one of the tar plugins I can't recall anymore; but it has the same "packaging == ? then setFile() else attach()") that actually makes it use the RPM as the primary artifact

Reply:
Here's a stackoverflow post talking about it https://stackoverflow.com/a/40790122/2312317 it's a few years old but it does a much better job explaining than I think I was doing